app/routes/login.tsx

```diff
@@ -24,107 +23,72 @@ export const links: LinksFunction = () => {
-              }
-            />
-            {fieldHasErrors(actionData, "username") ? (
-              <p
-                className="form-validation-error"
-                role="alert"
-                id="username-error"
-              >
-                {fieldFirstMessage(actionData, "username")}
-              </p>
-            ) : null}
-          </div>
-          <div>
-            <label htmlFor="password-input">Password</label>
-            <input
-              id="password-input"
-              name="password"
-              defaultValue={actionData?.fields?.password}
-              type="password"
-              aria-invalid={fieldHasErrors(actionData, "password")}
-              aria-errormessage={
-                fieldHasErrors(actionData, "password")
-                  ? "password-error"
-                  : undefined
-              }
-            />
-            {fieldHasErrors(actionData, "password") ? (
-              <p
-                className="form-validation-error"
-                role="alert"
-                id="password-error"
-              >
-                {fieldFirstMessage(actionData, "password")}
-              </p>
-            ) : null}
-          </div>
-          <div id="form-error-message">
-            {actionData?.errors.length ? (
-              <p className="form-validation-error" role="alert">
-                {actionData.errors.map(({ message }) => message).join(", ")}
-              </p>
-            ) : null}
-          </div>
-          <button type="submit" className="button">
-            Submit
-          </button>
+        <Form schema={registrationSchema} hiddenFields={["redirectTo"]}>
+          {({ Field, Errors, Button }) => (
+            <>
+              <Field name="loginType">
+                {({ Errors }) => (
+                  <fieldset>
+                    <legend className="sr-only">Login or Register?</legend>
+                    <label>
+                      <input
+                        type="radio"
+                        name="loginType"
+                        value="login"
+                        defaultChecked={
+                          !actionData?.fields?.loginType ||
+                          actionData?.fields?.loginType === "login"
+                        }
+                      />{" "}
+                      Login
+                    </label>
+                    <label>
+                      <input
+                        type="radio"
+                        name="loginType"
+                        value="register"
+                        defaultChecked={
+                          actionData?.fields?.loginType === "register"
+                        }
+                      />{" "}
+                      Register
+                    </label>
+                    <Errors />
+                  </fieldset>
+                )}
+              </Field>
+              <Field name="username" />
+              <Field name="password">
```

Comment:
```suggestion
              <Field name="password" type="password" />
```

This will achieve the same as passing the children below.